packages/core-commands/src/site-editor-navigation-commands.js

```diff
@@ -42,8 +43,16 @@ const getNavigationCommandLoaderPerPostType = ( postType ) =>
 					: {
 							per_page: -1,
 					  };
+				const results = getEntityRecords( 'postType', postType, query );
+				/*
+				 * wp_template and wp_template_part endpoints do not support per_page or orderby parameters.
+				 * We need to sort the results based on the search query to avoid removing relevant
+				 * records below using .slice().
+				 */
 				return {
-					records: getEntityRecords( 'postType', postType, query ),
+					records: supportsSearch
+						? results
+						: orderEntityRecordsBySearch( results || [], search ),
```

Comment:
This will return a different new array on each call. The `mapSelect` can be called multiple times when the store updates, but records here might be unchanged; it will cause unnecessary renderers because a new array reference is returned on each call.

It's better to do similar operations outside of the `mapSelect` and memoize when needed.

Reply:
Thanks @Mamaduka! I've just pushed a commit for that here https://github.com/WordPress/gutenberg/pull/53286/commits/b7d5f290c9e1dcf13340f112c1f0ff430b10a256